packages/store-ui/src/molecules/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import type { PropsWithChildren } from 'react'
+
+import Button from '../../atoms/Button'
+import Input from '../../atoms/Input'
+
+interface ActionButtonProps {
+  actionOption: 'dec' | 'inc'
+  disabled: boolean
+  ariaLabel: string
+  onClick: (currentValue: number) => void
+}
+
+const ActionButton = ({
+  children,
+  actionOption,
+  disabled,
+  ariaLabel,
+  onClick,
+}: PropsWithChildren<ActionButtonProps>) => (
+  <Button
+    type="button"
+    data-store-button={actionOption}
+    onClick={(event) => onClick(parseInt(event.currentTarget.value, 10))}
+    disabled={disabled}
+    aria-label={ariaLabel}
+    aria-disabled={disabled}
+  >
+    {children}
+  </Button>
+)
+
+export interface NumberInputProps {
```

Comment:
If we want to follow all specs from the `spin button,` we should receive all accessibility properties via Props. I don't know if we're going to allow it.

For example: 
```ts
interface Props {
  ...
  decreaseButtonProps:  {
    ...
    'aria-label'?: string // just an example for the left button props
  }
  'aria-valuenow'?: string | value // another example for the spinbutton itself
  ...
}
```